internal/config/options.go

```diff
@@ -161,11 +162,11 @@ func WithOpenShiftRoutesAvailability(os openshift.RoutesAvailability) Option {
 	}
 }
 
-func WithLabelFilters(labelFilters []string) Option {
+func WithSetFilters(setFilters []string, setType string) Option {
```

Comment:
I think rather than having a single method for both label and annotation, I would prefer a structure that looks like this:
```
WithLabelFilters(filters []string) Option
WithAnnotationFilters(filters []string) Option
```

both of these methods would call some shared function that does the logic for constructing the regex and filter list. That way we don't need to a potentially non-exhaustive switch and a caller is explicit about which config filter is being set.